dss/util/auth/authorize.py

```diff
@@ -46,9 +46,8 @@ def assert_required_parameters(self, provided_params: dict, required_params: lis
         """
         for param in required_params:
             if param not in provided_params:
-                title = "Missing Security Paramters"
-                err = f'Missing parameters within {provided_params.keys()}, unable to locate {param},'
-                raise DSSException(500, title, err)
+                err = f'Authorization token is missing claim {param} from claims: {provided_params.keys()}'
```

Comment:
this method is to generally check if a keyword exists within a dict of parameters, would 401 be appropirate in the case were we are using it to inspect if `griups` are present within some `kwargs`?

Reply:
I raise the 401 out of an attempt for consistency with `dss/util/security.py` and the `get_token_email` method there, which raises a 401 Unauthorized. The idea there is, if a parameter is missing from a security token, it's a security issue, not an internal server issue.

It's a good point you raise, though. I think we need to distinguish between a general method (assert parameters XYZ are in a given dict) raising a 500 error and a token-specific method (assert parameters XYZ are in the token) raising a 401 Unauthorized due to an invalid JWT.

Reply:
For more context: the integration tests were failing due to a missing email claim, from a test that was intentionally assembling malformed headers to ensure they raised 401 unauthorized errors. But those tests hit the security decorators first, which checked for email claims to see if the request was coming from an admin, which hit this method, which raised a 500 instead of a 401.

Reply:
Commit bf79ff0 implements a new `_assert_required_token_parameters()` method for the TokenMixin, which can be used to raise a 401. The existing `assert_required_parameters()` was restored to its old behavior of raising a 500. The new method calls the old method under the hood, catches the 500 exception, and raises a new 401 exception.